llvm/lib/CodeGen/SelectionDAG/FunctionLoweringInfo.cpp

```diff
@@ -69,6 +69,20 @@ static ISD::NodeType getPreferredExtendForValue(const Instruction *I) {
       NumOfSigned += CI->isSigned();
       NumOfUnsigned += CI->isUnsigned();
     }
+    if (const auto *CallI = dyn_cast<CallInst>(U)) {
+      for (unsigned op = 0, Eop = CallI->arg_size(); op < Eop; ++op) {
+        if (CallI->getArgOperand(op) == I) {
+          const AttributeList &PAL = CallI->getAttributes();
+          AttributeSet Attrs = PAL.getParamAttrs(op);
+          if (Attrs.hasAttributes()) {
+            for (const auto &Attr : Attrs) {
```

Comment:
You should be calling `hasAttribute()` on the AttributeSet, not iterate over all attributes inside it and check each individually.